File: anabeesh/src/main/java/com/cs18/anabeesh/muhammad/di/activity/ActivityModule.java
package com.cs18.anabeesh.muhammad.di.activity;

import android.app.Activity;
import android.content.Context;

import com.cs18.anabeesh.muhammad.ui.home.HomeScreen;

import dagger.Module;
import dagger.Provides;

/**
 This class is responsible for providing the requested objects to {@link ActivityScope} annotated classes
 */

@Module
public class ActivityModule {

    private final Activity activity;

    public ActivityModule(Activity activity) {this.activity = activity;}

    @ActivityScope
    @Provides
    Activity provideActivity() {
        return activity;
    }

    @ActivityScope
    @Provides
    @ForActivity
    Context provideActivityContext() {
        return activity;
    }

    @ActivityScope
    @Provides
    HomeScreen providesHomeScreen() {
        return (HomeScreen) activity;
    }
}

File: quora/src/main/java/com/cs18/quora/di/activity/ActivityModule.kt
package com.cs18.quora.di.activity

import android.app.Activity
import android.content.Context
import dagger.Module
import dagger.Provides

/**
 * This class is responsible for providing the requested objects for [ActivityScope] objects
 */

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    @ActivityScope
    fun providesActivity(): Activity = activity

    @ActivityScope
    @Provides
    @ForActivity
    fun providesActivityContext(): Context = activity
}